### ISSUE TITLE
[REQUEST] Migrate kubernetes-incubator/kube-aws to kubernetes-sigs

### ISSUE THREAD

**Issue by dominicgunn** [kubernetes/org#2004]
### New Repo, Staging Repo, or migrate existing
Migrate existing repository.

### Requested name for new repository
kube-aws

### Which Organization should it reside
kubernetes-sigs

### If not a staging repo, who should have admin access
dominicgunn, mumoshu

### If not a staging repo, who should have write access
dominicgunn, mumoshu

### If not a staging repo, who should be listed as approvers in OWNERS
dominicgunn, mumoshu

### If not a staging repo, who should be listed in SECURITY_CONTACTS
dominicgunn, mumoshu

### What should the repo description be
 A command-line tool to declaratively manage Kubernetes clusters on AWS 

### What SIG and subproject does this fall under in sigs.yaml
This is an existing subproject under the cluster lifecycle sig (https://github.com/kubernetes/community/tree/master/sig-cluster-lifecycle#kube-aws).

### Approvals

@neolit123 
@justinsb.

### Additional context for request

I wanted to open a discussion about one of the last remaining projects in the kubernetes-incubator organization, https://github.com/kubernetes-incubator/kube-aws. Admittedly, external-storage still exists there but it's readme links out to a repository under kubernetes-sigs. We want to help you guys be able to clean up and properly close down the incubation organization!

For a little bit of background, kube-aws is a pretty old project (started in 2015, incubated in 2017-03-15) originally sponsored by thockin & championed by Mike Danese, though I doubt at this point either will remember doing so. We were supposed to leave incubation sometime in 2018, and if I'm honest I joined the contribution team for kube-aws a little after that, so I'm not too sure what happened. We're obviously quite a small project, but we're still pretty active with a decent contribution base (see recent prs here: https://github.com/kubernetes-incubator/kube-aws/pulls?q=is%3Apr+is%3Aclosed).

Per discussion with @neolit123 I started an active discussion on the SIG cluuster lifecycle mailing list (https://groups.google.com/u/1/g/kubernetes-sig-cluster-lifecycle/c/2CfgHs-3rk8).

I'm also not sure if this is the right context, but it would be fantastic if we could both grab membership to the kubernetes-sig repository too, we've been championing this project used by several companies for years.

**Comment by neolit123** [comment]
this has been in discussion for a while now.

i have no historical context why kube-aws's readme says its maintained by SIG AWS, but on paper (under k/community) SIG CL is the owner and who added that there.

we might as well move the repository under k-sigs at this point.
so i'm +1 to do that.

if we do, please:
- include an OWNERS_ALIASES file that has the SIG leads and add those in the OWNERS file:
https://github.com/kubernetes-sigs/cluster-api/blob/master/OWNERS#L4-L6
https://github.com/kubernetes-sigs/cluster-api/blob/master/OWNERS_ALIASES#L3-L9
- remove this section from the readme.
- the k/org maintainers might request more changes.

thank you

**Comment by nikhita** [comment]
@neolit123 do you also want SCL leads to have admin access?

**Comment by neolit123** [comment]
i that is the standard practice, so yes i guess.

we also need to update existing links to the repository - e.g.:
https://github.com/kubernetes/community/tree/master/sig-cluster-lifecycle#kube-aws

**Comment by nikhita** [comment]
@dominicgunn Can you ping back here once the OWNERS files have been updated? The GitHub Admin team can take care of the rest :)

**Comment by dominicgunn** [comment]
Hey @nikhita, 

I've gone to start making the changes, but I'm getting shouted at as (despite already being in the owners file), I'm not a member of the kubernetes-incubator org.  (https://github.com/kubernetes-incubator/kube-aws/pull/1888/files)

I think I should be able to merge despite that, but just wanted to call it out incase it was problematic.

**Comment by nikhita** [comment]
@dominicgunn np, let's do this post migration :)

/assign

**Comment by nikhita** [comment]
One more hiccup -- @dominicgunn @mumoshu, you need to be members of the @kubernetes-sigs org before we can migrate the repo. Being a member is important because only members can be added to GitHub teams and be granted admin access.

The instructions about requesting membership are here - https://github.com/kubernetes/community/blob/master/community-membership.md#member

> I've gone to start making the changes, but I'm getting shouted at as (despite already being in the owners file), I'm not a member of the kubernetes-incubator org.

You could approve changes before because you are an outside collaborator for the `kube-aws` repo. All members and outside collaborators can lgtm and approve...but going forward, we are not allowing outside collaborators on repos anymore.

We recently added a new prow plugin which checks that OWNERS files contain org members (that's why the bot complains in https://github.com/kubernetes-incubator/kube-aws/pull/1888).

So, next steps would be:

- @dominicgunn @mumoshu need to create membership requests
- all outside collaborators will be removed post migration, so if anyone still wants to have access, they need to create membership requests too (I'd suggest checking the list of collaborators today and reaching out to them, if needed)

**Comment by dominicgunn** [comment]
Ok, membership request opened (https://github.com/kubernetes/org/issues/2005), and I've pinged both @neolit123 and @justinsb regarding sponsorship. I'll reach out to existing collaborators as required.

At this point, is then anything else you need from me or are we just waiting for the 👍 from sponsors @nikhita?

**Comment by dominicgunn** [comment]
I've also updated the [OWNERS](https://github.com/kubernetes-incubator/kube-aws/blob/master/OWNERS), [OWNERS_ALIASES](https://github.com/kubernetes-incubator/kube-aws/blob/master/OWNERS_ALIASES), and [README.md](https://github.com/kubernetes-incubator/kube-aws/blob/master/README.md) per request.

**Comment by nikhita** [comment]
> At this point, is then anything else you need from me or are we just waiting for the +1 from sponsors @nikhita?

Just waiting for +1s and then I'll migrate it! :rocket: 

I know it's a lot of back and forth, so thanks for your patience on this. :)

**Comment by mumoshu** [comment]
I didn't have the access to the kubernetes-sigs org so I've submitted my membership request at #2007 😃 
I wonder if it needs to be handled as well befobe finishing the migration?

**Comment by neolit123** [comment]
i'm putting this on hold until the next SIG CL meeting.
we had a discussion in the SIG CL slack channel and @timothysc expressed some concerns. i had the same originally, when this discussion started in 2019 (?). but given there is ownership of SIG CL (on paper, as per k/community sigs.yaml), it seemed like a natural move from incubator to k-sigs.

https://kubernetes.slack.com/archives/C13J86Z63/p1594830845241300

/hold

btw @nikhita i wasn't able to determine when kube-aws was added as a sig-cl sub-project and i started digging where it came from:

> i think it was added in this PR based on some automation around owner files:
> https://github.com/kubernetes/community/pull/1674

cc @spiffxp

**Comment by mrbobbytables** [comment]
Following up - It was decided to spin kube-aws out of incubator into its own project.
ref: https://groups.google.com/u/1/g/kubernetes-sig-contribex/c/YhFV68db9t8

Is there any other next steps before we can archive the current repo?

**Comment by neolit123** [comment]
there are pending questions on the mailing list(s) here:
https://groups.google.com/forum/#!topic/kubernetes-sig-cluster-lifecycle/zNHDdCUVh8w
contribex is on CC and one of the maintainers asked some questions.

@nikhita should also have some context about this.

let's close this ticket as it is no longer the plan forward.
/close
